### ISSUE TITLE
phpcs result depending on SlevomatCodingStandard.Files.TypeNameMatchesFileName configuration

### ISSUE THREAD

**Issue by BigCharlie** [slevomat/coding-standard#163]
Hi,

I've just met strange behavior of subject's rule. When trying to configure my namespaces, I found that:

```xml
<rule ref="SlevomatCodingStandard.Files.TypeNameMatchesFileName">
    <properties>
        <property name="rootNamespaces" type="array" value="tests/src=>X\Tests,tests=>X\Tests,"/>
    </properties>
</rule>
```

works as expected. But this:

```xml
<rule ref="SlevomatCodingStandard.Files.TypeNameMatchesFileName">
    <properties>
        <property name="rootNamespaces" type="array" value="tests=>X\Tests,tests/src=>X\Tests,"/>
    </properties>
</rule>
```
results in an error. There's an example of configuration in documentation (README.md, line 180, I don't know how to link better, sorry), but it's not mentioning the importance of values order.

Expected behavior
-----------------

Sniff should work regardless on configuration values order.

composer configuration
"slevomat/coding-standard": "^3.0"

**Comment by ondrejmirtes** [comment]
I'd just update the README to reflect this. For example autoloading configuration works the same way, the first path that matches wins...

**Comment by BigCharlie** [comment]
The only argument against is simplifying the configuration. I'm not able to find any better argument.

**Comment by kukulich** [comment]
Thanks for your report.

It should be fixed in https://github.com/slevomat/coding-standard/commit/d9e6e33cddd3dea80242bde788c1f22ca5cfa1a0
